core/domain/stats_services_test.py

```diff
@@ -1121,3 +1123,304 @@ def test_only_sample_answers_in_main_shard_returned(self):
+
+    def _record_answer(
+            self, answer, exp_id=TEXT_INPUT_EXP_ID, state_name=INIT_STATE_NAME):
+        exploration = exp_services.get_exploration_by_id(exp_id)
+        interaction_id = exploration.states[state_name].interaction.id
+        event_services.AnswerSubmissionEventHandler.record(
+            exp_id, exploration.version, state_name, interaction_id, 0, 0,
+            exp_domain.EXPLICIT_CLASSIFICATION, 'sid1', 10.0, {}, answer)
+
+    def _run_answer_summaries_aggregator(self):
+        ModifiedInteractionAnswerSummariesAggregator.start_computation()
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_CONTINUOUS_JOBS), 1)
+        self.process_and_flush_pending_tasks()
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_CONTINUOUS_JOBS), 0)
+
+    def _rerun_answer_summaries_aggregator(self):
+        ModifiedInteractionAnswerSummariesAggregator.stop_computation('a')
+        self._run_answer_summaries_aggregator()
+
+    def _rename_state(
+            self, new_state_name, exp_id=TEXT_INPUT_EXP_ID,
+            state_name=INIT_STATE_NAME):
+        exp_services.update_exploration(self.owner_id, exp_id, [{
+            'cmd': exp_domain.CMD_RENAME_STATE,
+            'old_state_name': state_name,
+            'new_state_name': new_state_name
+        }], 'Update state name')
+
+    def _change_state_interaction_id(
+            self, interaction_id, exp_id=TEXT_INPUT_EXP_ID,
+            state_name=INIT_STATE_NAME):
+        exp_services.update_exploration(self.owner_id, exp_id, [{
+            'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+            'state_name': state_name,
+            'property_name': exp_domain.STATE_PROPERTY_INTERACTION_ID,
+            'new_value': interaction_id
+        }], 'Update state interaction ID')
+
+    def _change_state_content(
+            self, new_content, exp_id=TEXT_INPUT_EXP_ID,
+            state_name=INIT_STATE_NAME):
+        exp_services.update_exploration(self.owner_id, exp_id, [{
+            'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+            'state_name': state_name,
+            'property_name': exp_domain.STATE_PROPERTY_CONTENT,
+            'new_value': {
+                'html': new_content,
+                'audio_translations': {},
+            }
+        }], 'Change content description')
+
+    def setUp(self):
+        super(AnswerVisualizationsTests, self).setUp()
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        user_services.create_new_user(self.owner_id, self.OWNER_EMAIL)
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.text_input_exploration = self.save_new_valid_exploration(
+            self.TEXT_INPUT_EXP_ID, self.owner_id, end_state_name='End')
+        self.text_input_exploration = self.save_new_valid_exploration(
```

Comment:
Wrong name? (Is this used at all anyway -- given that it wasn't detected?)

Reply:
Oops; removed all three of the variables since they aren't used.